src/compiler/core.ts

```diff
@@ -234,24 +236,19 @@ export function findMap<T, U>(array: readonly T[], callback: (element: T, index:
 
 /** @internal */
 export function contains<T>(array: readonly T[] | undefined, value: T, equalityComparer: EqualityComparer<T> = equateValues): boolean {
-    if (array) {
-        for (const v of array) {
-            if (equalityComparer(v, value)) {
+    if (array !== undefined) {
+        for (let i = 0; i < array.length; i++) {
+            if (equalityComparer(array[i], value)) {
                 return true;
             }
         }
     }
     return false;
 }
 
-/** @internal */
-export function arraysEqual<T>(a: readonly T[], b: readonly T[], equalityComparer: EqualityComparer<T> = equateValues): boolean {
-    return a.length === b.length && a.every((x, i) => equalityComparer(x, b[i]));
-}
-
 /** @internal */
 export function indexOfAnyCharCode(text: string, charCodes: readonly number[], start?: number): number {

```

Comment:
If this function is ever called in a hot path it would probably be more efficient to convert `charCodes` to a `Set` first to reduce the `O(text*codes)` to `O(text)`